packages/notebook/src/searchprovider.ts

```diff
@@ -266,43 +346,48 @@ export class NotebookSearchProvider extends SearchProvider<NotebookPanel> {
     this._query = query;
     this._filters = {
       output: false,
-      selectedCells: false,
+      selection: false,
       ...(filters ?? {})
     };
 
-    this._onSelectedCells = this._filters.selectedCells;
-    if (this._filters.selectedCells) {
-      this.widget.content.selectionChanged.connect(
-        this._onSelectionChanged,
-        this
-      );
-    }
+    this._onSelection = this._filters.selection;
+
+    const currentProviderIndex = this.widget.content.activeCellIndex;
 
     // For each cell, create a search provider
     this._searchProviders = await Promise.all(
-      cells.map(async cell => {
+      cells.map(async (cell, index) => {
         const cellSearchProvider = createCellSearchProvider(cell);
-        cellSearchProvider.stateChanged.connect(
-          this._onSearchProviderChanged,
-          this
-        );
 
         await cellSearchProvider.setIsActive(
-          !this._filters!.selectedCells ||
+          !this._filters!.selection ||
```

Comment:
`_filters` can potentially be `undefined` as [it is not initialized to anything](https://github.com/jupyterlab/jupyterlab/pull/14090/files#diff-40a4a43da20f655025356b02df592dd309f3f67562f6a66df22021d3f2771a7eR792), similar problem to what caused #14419. So using non-null assertion is not safe here. Potential alternatives:
1. Add check instead of non-null assertion at the current line
2. Explicitly initialize `_filters` to `null`, remove union with `undefined`, add check instead of non-null assertion at the current line
3. Initialize `_filters` to something like empty filter and remove union with undefined from its type

Reply:
Thank you for answering this and other questions, this helps me understand the codebase better